### DISCUSSION TITLE
<GeistProvider /> tag cannot be used with react-redux <Provider> tag?

### DISCUSSION THREAD

**Discussion by schwarzsky** [zeit-ui/react#512] [Q&A] [answered]
This is the App.js i've been working.

```
import logo from './logo.svg';
import './App.css';

import {useSelector, useDispatch} from 'react-redux'
import {increment, decrement} from './store/actions/'
import { GeistProvider, CssBaseline } from '@geist-ui/react';

function App() {
  const count = useSelector((state) => state.counter)
  const dispatch = useDispatch()

  return (
    <GeistProvider>
      <CssBaseline/>
      <div className="App">
        <header className="App-header">
          <h1>Count is {count}</h1>
          <button onClick={() => dispatch(increment(3))}>+</button>
          <button onClick={() => dispatch(decrement())}>-</button>
        </header>
      </div>
    </GeistProvider>
  );
}

export default App;
```

This is the index.js
```
import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';

import {GeistProvider, CssBaseline} from '@geist-ui/react'

import {Provider} from 'react-redux'
import store from './store'

ReactDOM.render(
    <Provider store={store}>
        <App />
    </Provider>,
  document.getElementById('root')
);
```

I tried use it on outside of <Provider> tag, inside of <Provider> tag but nothing changes. I get an error like `Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component.`

**Comment by Deep-Codes** [comment]
Works for me though

@schwarzsky try this repo I just created.

https://github.com/Deep-Codes/react-redux-geist-starter

**Reply by schwarzsky** [reply]
Hey, thanks for answer and example repo. I tried the same skeleton but still gives the same error. I created a repo if you wanna help. Thanks for help again. https://github.com/schwarzsky/redrux

**Reply by Deep-Codes** [reply]
I had 1 issue in that repo but then I updated Redux Tools and it worked.

![github](https://user-images.githubusercontent.com/61158210/114344071-cfd49e80-9b7c-11eb-8256-a78787d57952.gif)

**Reply by schwarzsky** [reply]
What do you mean by redux tools? the extension in browser or updated the dependecies with `npm install`?

**Reply by Deep-Codes** [reply]
browser one
My Error was this:
https://stackoverflow.com/questions/53514758/redux-typeerror-cannot-read-property-apply-of-undefined

**Reply by schwarzsky** [reply]
> browser one
> My Error was this:
> https://stackoverflow.com/questions/53514758/redux-typeerror-cannot-read-property-apply-of-undefined

i got same error when i uninstalled the extension. I installed it again and still getting the same Invalid Hook Call error. I am using brave browser like you, i tried to change this line: 

`const store = createStore(rootReducer, compose(applyMiddleware(...middlewares), window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : f => f))` 

but still getting the same error. Redux Dev Tools extension looks like v2.17.0

**Reply by Deep-Codes** [reply]
The last thing I do in such a situation is to delete and reinstall `node_modules` 🤔

**Reply by schwarzsky** [reply]
Ahahah, i am now installing chrome. I will try to look in chrome, if not works. I will try your way too. Thanks for help <3

**Reply by schwarzsky** [reply]
Not worked but really thanks for your help. <3 Can you tell what application are you using for creating screenrecorded gifs? And VS Code theme? Looks very nice.

**Reply by Deep-Codes** [reply]
I tried on chrome it works

I use Screenflow to record the video it exports in `.mp4` 

then I use FFmpeg script to convert to `.gif` 

You can find the script on my site: [FFmpeg Scripts](https://deepankarbhade.vercel.app/snippets/ffmpeg-scripts)

VsCode theme: Tokyo Night
Font: Dank Mono

**Comment by schwarzsky** [comment, marked as answer]
Problem was solved with using `npm remove @geist-ui/react` and installed with  `yarn add @geist-ui/react`. Thanks all. <3

**Reply by Gounlaf** [reply]
Reported https://github.com/geist-org/react/issues/519 cause using yarn and npm should works.
